src/nrfx/hal/nrf_egu.c

```diff
@@ -0,0 +1,176 @@
+/** Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Note that the function prototypes are taken from the NRFx HAL
+ */
+
+ #if 1
+
+#include "hal/nrf_egu.h"
+#include "bs_tracing.h"
+#include "NRF_EGU.h"
+#include <kernel.h>
+#include <irq_ctrl.h>
+
+uint32_t nrf_egu_channel_count(NRF_EGU_Type const *p_reg)
+{
+	return 16;
+}
+
+static int reg_to_irq(NRF_EGU_Type *p_reg)
+{
+	extern NRF_EGU_Type NRF_EGU_regs[6];
+	size_t raw_offset = (size_t)p_reg - (size_t)(&NRF_EGU_regs[0]);
```

Comment:
I would suggest using 'offsetof' instead

Reply:
ofsetof works on structs and unions, and NRF_EGU_regs is an array, In this part of code, I have an object, I know it is in this array, and I want to know under what index it is located. I could compare the address with all elements of this array, but i figured out that I can just subtract the address value and divide by the size of single object